src/js/layers/CartoLayerBase.js

```diff
@@ -0,0 +1,358 @@
+    let right = cartoCSS.match(/[^:\]]+(?=;)/g);
+    let left = cartoCSS.match(/.+?(?=:)/g);
+    right = right.map(function(string) { return string.trim(); });
+    left = left.map(function(string) { return string.trim(); });
+
+    // Creating an object out of the cartoCSS
+    left.forEach((leftProp, index) => {
+      if(cartoObj[leftProp] === undefined) {
+        cartoObj[leftProp] = right[index];
+      }
+    });
+
+    switch(type) {
+      case 'Point':
+        cartoObj['marker-opacity'] = cartoObj['marker-opacity'] ? cartoObj['marker-opacity'] : 1;
+        esriObj.color = this.convertHex(cartoObj['marker-fill'], cartoObj['marker-opacity']);
+        esriObj.size = cartoObj['marker-width'];
+        esriObj.style = 'STYLE_CIRCLE';
+        esriObjLineSymbol.width = cartoObj['marker-line-width'];
+        break;
+      case 'MultiPolygon':
+        esriObj.color = this.convertHex(cartoObj['polygon-fill'], cartoObj['polygon-opacity']);
+        esriObjLineSymbol.color = this.convertHex(cartoObj['line-color'], cartoObj['line-opacity']);
+        esriObjLineSymbol.width = cartoObj['line-width'];
+        break;
+      case 'Polyline':
+        // TODO but chances are we are not going to see any polylines
+        break;
+    }
+    return { esriObj, esriObjLineSymbol };
+  },
+
+  convertHex: function(hex, opacity){
+    const result = [];
+    hex = hex.replace('#', '');
+    hex = hex.length === 3 ? hex.repeat(2) : hex;
+
+    const red = parseInt(hex.substring(0, 2), 16);
+    const green = parseInt(hex.substring(2, 4), 16);
+    const blue = parseInt(hex.substring(4, 6), 16);
+
+    result.push(red, green, blue, opacity);
+    return result;
+  },
+
+  /**
+   * The main query function of the CartoDBLayer
+   * @param {string} queryString
+   */
+  query: function(cartoQuery, cartoTemplate, layerNumber, layerIndex, cartocss, layerName, cartoLayers, resolve, reject) {
+    var _url = urls.cartoDataEndpoint(this.cartoUser, cartoQuery, this.cartoApiKey);
+    var esriJsonLayer = [];
+    request.id = 2;
+
+    request(_url, {timeout: 5000}).then((data) => {
+
+      var geojson = dojoJSON.parse(data);
+      const meta = this.setMetadataFields(geojson.features[0].properties, layerNumber);
+      const { esriObj, esriObjLineSymbol } = this.processCartoCSS(cartocss, geojson.features[0].geometry.type);
+      this.setParameters(geojson.features[0].geometry.type, esriObj, esriObjLineSymbol);
+      const esriJson = geojsonUtil.geojsonToArcGIS(geojson);
+
+      esriJson.forEach(feature => {
+        if (feature.geometry) {
+          var graphic = new Graphic(feature);
+          //project geometry to web mercator if needed
+
+          if (graphic.geometry.spatialReference.wkid === 4326){
+            graphic.setGeometry(
+              webMercatorUtils.geographicToWebMercator(graphic.geometry)
+            );
+          }
+
+          // Set the symbolDictionary depending on the geometry type
+          if (!!this.symbolDictionary) {
```

Comment:
This part of the code has been copied directly from odoenet's carto implementation